packages/core/src/StateNode.ts

```diff
@@ -976,11 +976,22 @@ class StateNode<
   private getExternalReentryNodes(
```

Comment:
A test case should be added with this change because I'm not sure what exactly is being fixed here.

Reply:
I will add one.

Reply:
I added three tests. The third one I'm not 100% is the correct behaviour. The documentation says when it's external it should exit, but it doesn't seem to be happening there.

Reply:
> The third one I'm not 100% is the correct behaviour. The documentation says when it's external it should exit, but it doesn't seem to be happening there.

Do you refer to the first test case and to the fact that this `A` state is not being exited during this transition?
https://github.com/statelyai/xstate/blob/111820100772bf5da515edc3d6fc40931a7d358f/packages/core/test/actions.test.ts#L532

Reply:
I see that we've discussed this in the other thread [here](https://github.com/statelyai/xstate/pull/3623#discussion_r985128206) but the `exit` has not been added to this particular state - this is perhaps to make the test pass. Would this be correct?